File: domains/system/api/oplog.ts
import { ApiEntry } from "@/client"
import type { ApiResponse, PaginatedResult } from "@/common"
import type { OpLogListQuery } from "../types"
import type { OpLog } from "../models"

const opLogApi = new ApiEntry("/system/oplogs")

export async function listOpLogs(query: OpLogListQuery): ApiResponse<PaginatedResult<OpLog>> {
    return await opLogApi.get("/", query)
}

File: domains/system/api/oplog.go
package api

import (
	"fmt"

	"bilingo/domains/system/service"
	"bilingo/domains/system/types"
	"bilingo/server"
	"bilingo/server/auth"

	"github.com/gofiber/fiber/v2"
)

var OpLogApi = server.NewApiEntry("/system/oplogs", auth.UseAuth)

func init() {
	OpLogApi.Get("/", auth.RequireAuth, listOpLogs)
}

func listOpLogs(ctx *fiber.Ctx) error {
	var query types.OpLogListQuery
	if err := ctx.QueryParser(&query); err != nil {
		return server.Error(ctx, 400, fmt.Errorf("malformed query: %w", err))
	}

	result, err := service.ListOpLogs(ctx.UserContext(), query)
	if err != nil {
		return server.Error(ctx, 500, err)
	}

	return server.Success(ctx, result)
}
